core/src/env/srml/srml_only/impls.rs

```diff
@@ -199,21 +198,20 @@ where
     ) -> Result<U, CallError> {
         let callee = callee.encode();
         let value = value.encode();
-        unsafe {
-            let success = sys::ext_call(
+        let result = unsafe {
+            sys::ext_call(
                 callee.as_ptr() as u32,
                 callee.len() as u32,
                 gas,
                 value.as_ptr() as u32,
                 value.len() as u32,
                 input_data.as_ptr() as u32,
                 input_data.len() as u32,
-            );
-            if success == 0 {
-                U::decode(&mut &read_scratch_buffer()[..]).map_err(|_| CallError)
-            } else {
-                Err(CallError)
-            }
+            )
+        };
```

Comment:
I've made a quick PR into this branch with this block of code extracted to a method #172. Because only the last 3 lines of each of these methods are interesting.